test/augmentation/test_base.py

```diff
@@ -51,27 +51,45 @@ def test_forward_params(self, p, p_batch, same_on_batch, num, seed, device, dtyp
             assert "batch_prob" in output
             assert len(output['degrees']) == output['batch_prob'].sum().item() == num
 
-    def test_forward(self, device, dtype):
+    @pytest.mark.parametrize('keepdim', (True, False))
+    def test_forward(self, device, dtype, keepdim):
         torch.manual_seed(42)
         input = torch.rand((12, 3, 4, 5), device=device, dtype=dtype)
-        expected_output = input.unsqueeze(dim=1)[..., :2, :2]
-        augmentation = _BasicAugmentationBase(p=.3, p_batch=1.)
+        expected_output = input[..., :2, :2] if keepdim else input.unsqueeze(dim=0)[..., :2, :2]
+        augmentation = _BasicAugmentationBase(p=.3, p_batch=1., keepdim=keepdim)
         with patch.object(augmentation, "apply_transform", autospec=True) as apply_transform, \
                 patch.object(augmentation, "generate_parameters", autospec=True) as generate_parameters, \
-                patch.object(augmentation, "transform_tensor", autospec=True) as transform_tensor:
+                patch.object(augmentation, "transform_tensor", autospec=True) as transform_tensor, \
+                patch.object(augmentation, "__check_batching__", autospec=True) as check_batching:
 
             generate_parameters.side_effect = lambda shape: {
                 'degrees': torch.arange(0, shape[0], device=device, dtype=dtype)
             }
-            transform_tensor.side_effect = lambda x: x.unsqueeze(dim=1)
+            transform_tensor.side_effect = lambda x: x.unsqueeze(dim=0)
             apply_transform.side_effect = lambda input, params: input[..., :2, :2]
+            check_batching.side_effect = lambda input: None
             output = augmentation(input)
             assert output.shape == expected_output.shape
             assert_allclose(output, expected_output)
 
 
 class TestAugmentationBase2D:
 
+    @pytest.mark.parametrize('input_shape, in_trans_shape', [
+        ((2, 3, 4, 5), (2, 3, 3)),
+        ((3, 4, 5), (3, 3)),
+        ((4, 5), (3, 3)),
+        pytest.param((1, 2, 3, 4, 5), (2, 3, 3), marks=pytest.mark.xfail),
+        pytest.param((2, 3, 4, 5), (1, 3, 3), marks=pytest.mark.xfail),
+        pytest.param((2, 3, 4, 5), (3, 3), marks=pytest.mark.xfail),
+    ])
+    def test_check_batching(self, device, dtype, input_shape, in_trans_shape):
```

Comment:
I added a test for __check_batching__ here.

Reply:
Now that I think the case `(2, 3, 4, 5)` and `(1, 3, 3)` is pretty legit, as the batch dim is broadcasted. Should I make an exception for it?